napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -0,0 +1,255 @@
+            self._table.setRowCount(1)
+            self._table.setColumnCount(3)
+            self._table.setHorizontalHeaderLabels(['Action', 'Keybinding'])
+            self._table.verticalHeader().setVisible(False)
+
+            self._table.setColumnHidden(2, True)
+            item = QTableWidgetItem('No key bindings')
+            item.setFlags(item.flags() & ~Qt.ItemIsEditable)
+            self._table.setItem(0, 0, item)
+
+    def _set_keybinding(self, event):
+        # event is the index of the event.
+
+        current_layer_text = self.layer_combo_box.currentText()
+        layer_actions = self.key_bindings_strs[current_layer_text]
+
+        # get the current item from shortcuts column
+        item1 = self._table.currentItem()
+        new_shortcut = item1.text()
+
+        # get the action name
+        current_action = self._table.item(event, 2).text()
+
+        replace = True
+        for row, (action_name, action) in enumerate(layer_actions.items()):
+            shortcuts = action_manager._shortcuts.get(action_name, [])
+
+            if new_shortcut in shortcuts:
+                # shortcut is here (either same action or not), don't replace in settings.
+                replace = False
+                if action_name != current_action:
+                    # the shortcut is saved to a different action, show message.
+                    # pop up window for warning.
+                    message = trans._(
+                        f"The keybinding <b>{new_shortcut}</b> "
+                        + f"is already assigned to <b>{action.description}</b>; change or clear "
+                        + f"that shortcut before assigning <b>{new_shortcut}</b> to this one."
+                    )
+
+                    self._warn_dialog = KeyBindWarnPopup(
+                        parent=self,
+                        text=message,
+                    )
+
+                    self._warn_dialog.exec_()
+
+                    # get the original shortcut
+                    current_shortcuts = list(
+                        action_manager._shortcuts.get(current_action, {})
+                    )
+                    # reset value in table to value stored in action manager.
+                    item1.setText(current_shortcuts[0])
+
+                    break
+
+            # else:
+
+        if replace is True:
+
+            # this shortcut is not taken, can set it and save in settings
+            # how are we doing this? saving in settings and then that triggers to update the action manager?
+            # right now I'm updating the action manager, and settings will be saved after a trigger as with
+            # all the other widgets.
+
+            #  Bind new shortcut to the action manager
+            action_manager.unbind_shortcut(current_action)
+            action_manager.bind_shortcut(current_action, new_shortcut)
+
+            # Save to settings here temporarily (probably won't do this here)
```

Comment:
+1, we can probably ask the action manager for all shortcuts except the default. I'll look into that.